src/type_translator.ts

```diff
@@ -59,11 +48,23 @@ function isClutzDts(sourceFile: ts.SourceFile): boolean {
 }
 
 /**
- * Returns true if the symbol is defined in a Clutz-generated d.ts file.
+ * typeValueConflictHandled returns true for symbols whose type/value conflict is handled outside of
+ * tsickle.
+ *
+ * This covers two cases:
+ *
+ * - symbols provided by Clutz. Given that Closure has a merged type/value namespace, apparent
+ *   type/value conflicts on the TypeScript level are actually fine.
+ * - builtin lib*.d.ts symbols, such as "Array", which are considered Closure-compatible. Note that
+ *   we don't actually enforce that the types are actually compatible, but mostly just hope that
+ *   they are due to being derived from the same HTML specs.
  */
-export function isClutzType(symbol: ts.Symbol): boolean {
+export function typeValueConflictHandled(symbol: ts.Symbol) {
+  // TODO(#1072): if the symbol comes from a tsickle-transpiled file, either .ts or .d.ts with
```

Comment:
I don't follow this comment.  Aren't tsickle output files always .js instead of .ts or .d.ts?

Reply:
I copied this from a comment authored by Evan in the other file. We might need to ask him, but I think he's talking about a symbol that looks like a type/value conflict because how tsickle generated `.closure.js` or `.externs.js`, which in turn was then consumed by Clutz.

Reply:
If he were here, I'd ask him to suggest a clarification, but I don't think we should block this PR on that.

Reply:
I tried to go through the cases of type/value conflicts here:
https://github.com/angular/tsickle/issues/1072#issuecomment-533894437
we now handle 1 and 2.  This comment was about 3 and 4.

The idea is that in TS-authored code with type/value conflicts, we could emit the type in JS with a mangled name like MyClass$type.  Due to separate compilation, for an import we only see the original .ts/.d.ts types, and not the emitted .js, so it's hard to know for a given import whether it is one where the name is mangled or not.

I have more info here
https://github.com/angular/tsickle/pull/1077
where I think the goal is mostly right (mangle the names) but the approach isn't right (how to tell when to mangle a name or not).